### ISSUE TITLE
resolve self_url from flask view name

### ISSUE THREAD

**Issue by jo-tham** [marshmallow-code/marshmallow-jsonapi#36]
thanks to @asteinlein, [pr 19](https://github.com/marshmallow-code/marshmallow-jsonapi/pull/19/files) introduced self_url to the schema via `Schema` `Meta`

```
...
class Meta:
    self_url = 'api/users/{}'
    self_url_kwargs = {'id': '<id>'}
    self_url_many = 'api/users'
```

it would be nice to be able to use route names to avoid hard-coding these urls, in the manner that relationship fields operate.

```
...
class Meta:
    self_view = 'user'
    self_view_kwargs = {'id': '<id>'}
    self_view_many = 'users'
```

but how to introduce this to the `Schema` without introducing flask dependency. 

perhaps on a case by case basis with `@post_dump`?
or have a `FlaskSchema` that implements this?

any suggestions on the most suitable approach @sloria & co? with some input on the general direction I'm glad to make a PR

**Comment by sloria** [comment]
This is already supported using `marshmallow_json.flask.Relationship`.

**Comment by asteinlein** [comment]
I'm back on the project where we're using marshmallow-jsonapi, so I figured I might as well fix this while I'm at it. Relationships have supported resolving URLs from views for a while, but not self links in the Meta options.

PR #51 implements this. New options `self_view`, `self_view_kwargs` and `self_view_many` were added to a Flask-specific schema, instead of changing the behavior of `self_url` et al. This follows the semantics from `marshmallow_json.flask.Relationship`, which has `related_view`, `self_view`, etc, instead of changing `related_url` etc.
